solc/CommandLineInterface.cpp

```diff
@@ -584,6 +587,37 @@ bool CommandLineInterface::parseLibraryOption(string const& _input)
 	return true;
 }
 
+//read file contents from m_sourceCodes into m_sourceJsons, which either are
+// a JSON-file as produced by
+// a) --combined-json -ast,compact-format
+// b) the standard-json output
+map<string, Json::Value const*> CommandLineInterface::parseAstFromInput()
+{
+	map<string, Json::Value const*> sourceJsons;
+	map<string, string> tmp_sources; //used to generate the onchainmetadata-hash
+	for (auto const& srcPair: m_sourceCodes)
+	{
+		Json::Value* ast = new Json::Value();
```

Comment:
```suggestion
		Json::Value ast;
```
Otherwise, this will create a dangling pointer that is not cleared.